### PR TITLE
Add Chinese conversion function by bridging SwiftyOpenCC

### PR DESCRIPTION
**Pull request by zonble**
A new item is added to the input method menu to toggle Chinese conversion on or off. Once the switch is on, the text input will be converted into Simplified Chinese.

The function bridge a Chinese conversion library [SwiftyOpenCC](https://github.com/ddddxxx/SwiftyOpenCC). Since the library has only pure Swift interface, there are some class for bridging Objective-C and Swift code.

### CODE REVIEW

File: Source/InputMethodController.mm [hunk 1]
```diff
@@ -218,32 +216,31 @@ - (NSMenu *)menu
...
 
-            NSMenuItem *dumpMenuItem = [[[NSMenuItem alloc] initWithTitle:NSLocalizedString(@"Dump Learning Data to Console", @"") action:@selector(dumpLearningDictionary:) keyEquivalent:@""] autorelease];
+            NSMenuItem *dumpMenuItem = [[NSMenuItem alloc] initWithTitle:NSLocalizedString(@"Dump Learning Data to Console", @"") action:@selector(dumpLearningDictionary:) keyEquivalent:@""];
             [menu addItem:dumpMenuItem];
         }
     }
     #endif //DEBUG
 
-    NSMenuItem *updateCheckItem = [[[NSMenuItem alloc] initWithTitle:NSLocalizedString(@"Check for Updates…", @"") action:@selector(checkForUpdate:) keyEquivalent:@""] autorelease];
+    NSMenuItem *chineseConvertionMenuItem = [[NSMenuItem alloc] initWithTitle:NSLocalizedString(@"Chinese Convertion", @"") action:@selector(toggleChineseConverter:) keyEquivalent:@"G"];
```
Comment (lukhnos) [inline Source/InputMethodController.mm]: Nit: Conversion.

Perhaps also consider `s/Convertion/Conversion/g` for properties, constants, and others?

File: Source/InputMethodController.mm [hunk 2]
```diff
@@ -75,6 +78,7 @@
 static NSString *const kComposingBufferSizePreferenceKey = @"ComposingBufferSize";
 static NSString *const kDisableUserCandidateSelectionLearning = @"DisableUserCandidateSelectionLearning";
 static NSString *const kChooseCandidateUsingSpaceKey = @"ChooseCandidateUsingSpaceKey";
+static NSString *const kChineseConvertionEnanledKey = @"ChineseConvertionEnanledKey";
```
Comment (lukhnos) [inline Source/InputMethodController.mm]: Nit: Enabled.

File: McBopomofo.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -821,6 +846,8 @@
 				ALWAYS_SEARCH_USER_PATHS = NO;
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CLANG_CXX_LANGUAGE_STANDARD = "gnu++0x";
+				CLANG_ENABLE_MODULES = YES;
+				CLANG_ENABLE_OBJC_ARC = YES;
```
Comment (lukhnos) [inline McBopomofo.xcodeproj/project.pbxproj:850]: 🎉

File: Source/Installer/AppDelegate.h [hunk 4]
```diff
@@ -34,20 +34,20 @@
...
 - (IBAction)cancelAction:(id)sender;
 
-@property (assign) IBOutlet NSButton *installButton;
-@property (assign) IBOutlet NSButton *cancelButton;    
-@property (assign) IBOutlet NSTextView *textView;
-@property (assign) IBOutlet NSWindow *progressSheet;
-@property (assign) IBOutlet NSProgressIndicator *progressIndicator;
+@property (weak) IBOutlet NSButton *installButton;
+@property (weak) IBOutlet NSButton *cancelButton;    
+@property (unsafe_unretained) IBOutlet NSTextView *textView;
```
Comment (lukhnos) [inline Source/Installer/AppDelegate.h:50]: Hmm I haven't done this exercise for a long time :/ any reason why `textView` can't be `weak` like others? (Same question for line 39.)

**Review by lukhnos — APPROVED** [review]
LGTM. Thank you!

File: Source/Installer/AppDelegate.h (hunk 4 above)
Comment (zonble) [reply Source/Installer/AppDelegate.h:50]: See https://developer.apple.com/library/archive/releasenotes/ObjectiveC/RN-TransitioningToARC/Introduction/Introduction.html#//apple_ref/doc/uid/TP40011226

    Which classes don’t support weak references?

    You cannot currently create weak references to instances of the following classes:

    NSATSTypesetter, NSColorSpace, NSFont, NSMenuView, NSParagraphStyle, NSSimpleHorizontalTypesetter, and NSTextView.

Comment (zonble) [reply Source/Installer/AppDelegate.h:50]: That is what happens if we change it to weak.

Screen Shot 2021-11-11 at 00 11 24

File: Source/InputMethodController.mm (hunk 1 above)
Comment (zonble) [reply Source/InputMethodController.mm]: Let me do it.

**Comment by ShikiSuen** [comment]
Is this already optimized for compiling against macOS 10.14 Mojave SDK?

**Pull request merged**